### PR TITLE
Set catalog when executing rule tests in transactions

### PR DESCRIPTION
**Pull request by rschlussel**
@martint this is a fixup for the commit to execute rule tests in transactions.  I accidentally left out the line for setting the catalog, which is necessary for TableScanNodes to work correctly.

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java [hunk 1]
```diff
@@ -143,6 +143,10 @@ public void matches(PlanMatchPattern pattern)
     {
         return transaction(transactionManager, accessControl)
                 .singleStatement()
-                .execute(session, transactionSessionConsumer);
+                .execute(session, session -> {
+                            session.getCatalog().ifPresent(catalog -> metadata.getCatalogHandle(session, catalog));
+                            return transactionSessionConsumer.apply(session);
+                        }
+                );
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java]: Nit: put this on the previous line

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java [hunk 2]
```diff
@@ -143,6 +143,10 @@ public void matches(PlanMatchPattern pattern)
     {
         return transaction(transactionManager, accessControl)
                 .singleStatement()
-                .execute(session, transactionSessionConsumer);
+                .execute(session, session -> {
+                            session.getCatalog().ifPresent(catalog -> metadata.getCatalogHandle(session, catalog));
```
Comment (martint) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java]: How does this set anything? Does metadata.getCatalogHandle have side-effects (ugh :/)?

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java [hunk 3]
```diff
@@ -143,6 +143,10 @@ public void matches(PlanMatchPattern pattern)
...
+                            return transactionSessionConsumer.apply(session);
+                        }
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java]: This should be aligned with the first column of `.execute`

Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java]:
```java
.execute(..., x -> {
    ...
});

Comment (rschlussel) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java]: fixed

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java (hunk 2 above)
Comment (rschlussel) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java]: yup, ugly side effects.

Comment (martint) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java]: Add a comment, then. It's not obvious at all.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java [hunk 4]
```diff
@@ -143,6 +143,9 @@ public void matches(PlanMatchPattern pattern)
     {
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java]: consider `s/tests/rule tests/` in the cmt msg.

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**